src/main/java/org/opentripplanner/transit/model/timetable/ScheduledTripTimesBuilder.java

```diff
@@ -71,59 +76,143 @@ public ScheduledTripTimesBuilder plusTimeShift(int delta) {
     return this;
   }
 
+  public int serviceCode() {
+    return serviceCode;
+  }
+
   public ScheduledTripTimesBuilder withServiceCode(int serviceCode) {
     this.serviceCode = serviceCode;
     return this;
   }
 
+  public int[] arrivalTimes() {
+    return arrivalTimes;
+  }
+
   public ScheduledTripTimesBuilder withArrivalTimes(int[] arrivalTimes) {
     this.arrivalTimes = deduplicator.deduplicateIntArray(arrivalTimes);
     return this;
   }
 
+  /** For unit testing, uses {@link TimeUtils#time(java.lang.String)}. */
+  public ScheduledTripTimesBuilder withArrivalTimes(String arrivalTimes) {
+    return withArrivalTimes(TimeUtils.times(arrivalTimes));
+  }
+
+  public int[] departureTimes() {
+    return departureTimes;
+  }
+
   public ScheduledTripTimesBuilder withDepartureTimes(int[] departureTimes) {
     this.departureTimes = deduplicator.deduplicateIntArray(departureTimes);
     return this;
   }
 
+  /** For unit testing, uses {@link TimeUtils#time(java.lang.String)}. */
+  public ScheduledTripTimesBuilder withDepartureTimes(String departureTimes) {
+    return withDepartureTimes(TimeUtils.times(departureTimes));
+  }
+
+  public BitSet timepoints() {
+    return timepoints == null ? EMPTY_BIT_SET : timepoints;
+  }
+
   public ScheduledTripTimesBuilder withTimepoints(BitSet timepoints) {
     this.timepoints = deduplicator.deduplicateBitSet(timepoints);
     return this;
   }
 
+  public Trip trip() {
+    return trip;
+  }
+
   public ScheduledTripTimesBuilder withTrip(Trip trip) {
     this.trip = trip;
     return this;
   }
 
+  public List<BookingInfo> dropOffBookingInfos() {
+    return dropOffBookingInfos == null ? List.of() : dropOffBookingInfos;
```

Comment:
```suggestion
    return Objects.requireNonNullElse(dropOffBookingInfos, List.of());
```

Reply:
In some cases using `Objects.requireNonNullElse(V, C)` makes the code more readable - it is used to assert that a value is indeed NOT null (it verify the default-value as well and throw an exception). It should not be used to replace the `?:` operator. So, in this case I think it should not be used. Note! The `Objects.requireNonNullElseGet()` is slightly better, to avoid the `List.of()` call, but `?:` is preferred.